data/src/test/java/io/homeassistant/companion/android/data/authentication/AuthenticationRepositoryImplSpec.kt

```diff
@@ -98,6 +97,29 @@ object AuthenticationRepositoryImplSpec : Spek({
             }
         }
 
+        describe("build bearer token"){
+            it("should return a valid bearer token"){
+                coEvery { localStorage.getString("url") } returns "https://demo.home-assistant.io/"
+                coEvery { localStorage.getString("access_token") } returns "ABCDEFGH"
+                coEvery { localStorage.getLong("expires_date") } returns 1547605320
+                coEvery { localStorage.getString("refresh_token") } returns "IJKLMNOPQRST"
+                coEvery { localStorage.getString("token_type") } returns "Bearer"
+
+                val token = runBlocking { repository.buildBearerToken() }
+
+                assertThat(token).isEqualTo("Bearer ABCDEFGH")
+            }
+            it("should throw an exception when no valid session"){
+                coEvery { localStorage.getString("url") } returns null
+                coEvery { localStorage.getString("access_token") } returns null
+                coEvery { localStorage.getLong("expires_date") } returns null
+                coEvery { localStorage.getString("refresh_token") } returns null
+                coEvery { localStorage.getString("token_type") } returns null
+
+                assertThatThrownBy { runBlocking { repository.buildBearerToken() } }
+            }
+        }
+
```

Comment:
You can put this test function inside 
describe("connected user with valid access token") 
describe("connected user with expired access token") 
describe("anonymous user")
You will not need to mock session storage (it's already done in the describe).
Do you see the point of putting mock in the describe ?

Reply:
Makes sense.